firebase-ai/src/main/kotlin/com/google/firebase/ai/type/ImagenEditingConfig.kt

```diff
@@ -0,0 +1,68 @@
+package com.google.firebase.ai.type
+
+import kotlinx.serialization.Serializable
+
+@PublicPreviewAPI
+public class ImagenEditingConfig(
+  public val image: ImagenInlineImage,
+  public val editMode: ImagenEditMode,
+  public val mask: ImagenInlineImage? = null,
+  public val maskDilation: Double? = null,
+  public val editSteps: Int? = null,
+) {
+  public companion object {
+    public fun builder(): Builder = Builder()
+  }
+
+  public class Builder {
+    @JvmField public var image: ImagenInlineImage? = null
+    @JvmField public var editMode: ImagenEditMode? = null
+    @JvmField public var mask: ImagenInlineImage? = null
+    @JvmField public var maskDilation: Double? = null
+    @JvmField public var editSteps: Int? = null
+
+    public fun setImage(image: ImagenInlineImage): Builder = apply { this.image = image }
+
+    public fun setEditMode(editMode: ImagenEditMode): Builder = apply { this.editMode = editMode }
+
+    public fun setMask(mask: ImagenInlineImage): Builder = apply { this.mask = mask }
+
+    public fun setMaskDilation(maskDilation: Double): Builder = apply {
+      this.maskDilation = maskDilation
+    }
+
+    public fun setEditSteps(editSteps: Int): Builder = apply { this.editSteps = editSteps }
+
+    public fun build(): ImagenEditingConfig {
```

Comment:
If both `image` and `editMode` cannot be null, move them as part of the constructor, and the `build()` method, so devs are forced to set them up.

Reply:
The sample Kotlin type-safe builder pattern does uses parameters if the data is required, see `a(href, init)` in https://kotlinlang.org/docs/type-safe-builders.html#full-definition-of-the-com-example-html-package